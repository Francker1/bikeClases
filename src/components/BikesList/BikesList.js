import React from 'react';
import classNames from 'classnames';

export default function BikesList({ className }) {
  return (
    <div className={classNames('bikes-list', className)}>
      Bikes List
      <ul>
        {[1, 2, 3, 4, 5, 6, 7, 8, 9, 10].map(i => (
          <li key={i}>{i}</li>
        ))}
      </ul>
    </div>
  );
}
