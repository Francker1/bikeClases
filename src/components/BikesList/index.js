import { connect } from 'react-redux';

import BikesList from './BikesList';

import { addToCart } from '../../store/actions';
import { getVisibleBikes } from '../../store/selectors';

function mapStateToProps(state) {
  return {
    bikes: getVisibleBikes(state.bikes, state.bikesFilter),
  };
}

function mapDispatchToProps(dispatch) {
  return {
    addToCart: itemId => dispatch(addToCart(itemId)),
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(BikesList);
