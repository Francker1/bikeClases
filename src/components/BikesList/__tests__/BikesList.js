import React from 'react';
import { shallow } from 'enzyme';

import BikesList from '../BikesList';

describe('BikesList', () => {
  const defaultProps = {
    addToCart: jest.fn(),
    bikes: [
      {
        id: 'bike1',
        name: 'mountain cf 7.0',
        image: 'url/to/image',
        price: 3999.99,
        hasStock: true,
      },
    ],
  };

  const render = props => shallow(<BikesList {...defaultProps} {...props} />);
  let wrapper;

  beforeEach(() => {
    wrapper = render();
  });

  it('should render BikesList component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should add a bike to cart', () => {
    const [bike] = defaultProps.bikes;
    const item = shallow(
      wrapper
        .find('.list')
        .props()
        .renderItem(bike),
    );
    item.find('.actions button').simulate('click');
    expect(defaultProps.addToCart).toHaveBeenCalledWith(bike.id);
  });
});
