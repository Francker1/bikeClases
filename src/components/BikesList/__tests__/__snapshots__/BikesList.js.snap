// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BikesList should render BikesList component 1`] = `
<div
  className="bikes-list"
>
  Bikes List
  <List
    className="list"
    items={
      Array [
        Object {
          "hasStock": true,
          "id": "bike1",
          "image": "url/to/image",
          "name": "mountain cf 7.0",
          "price": 3999.99,
        },
      ]
    }
    renderItem={[Function]}
  />
</div>
`;
